test/catRoutes.test.js

```diff
@@ -106,16 +118,23 @@ describe('PUT /cats/:id', function () {
             .send(falseBirthday)
             .set('Accept', "text/html; charset=utf-8")
             .expect('Content-Type', "text/html; charset=utf-8")
-            .expect(400, `Error: Invalid ageInYears parameter: "two"`, done)
+            .expect(400, `Error: Invalid ageInYears parameter: "two"`)
+            .end((err) => {
+                if (err) return done(err); 
+                done(); 
+            })    
     });  
 })
-//  Error: expected 400 "Bad Request", got 500 "Internal Server Error"
 
 describe('DEL /cats/:id', function () {
     it('respond with 204 No Content', function (done) {
         request(appTest)
             .delete('/cats/tmk60ux2b')
-            .expect(204, done)
+            .expect(204)
+            .end((err) => {
+                if (err) return done(err); 
+                done(); 
```

Comment:
is it possible to create a little utility function in the test file that avoids repeating this?

Reply:
I don't think so because mocha's done() is only recognised as a function when passed as the argument of a it() test case callback. I tried extracting it out as a global function variable before all the test cases and it wouldn't recognise it as a function. You might have had something else in mind though - please share!

Reply:
sounds like we should leave it alone in that case.